internal/rollout/device_selection/batch_sequence_selector.go

```diff
@@ -0,0 +1,462 @@
+package device_selection
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/flightctl/flightctl/internal/store/selector"
+	"github.com/flightctl/flightctl/internal/util"
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	DefaultSuccessThreshold = 90
+	Unknown                 = "Unknown"
+	ApplyingUpdate          = "ApplyingUpdate"
+	Rebooting               = "Rebooting"
+	ActivatingConfig        = "ActivatingConfig"
+	RollingBack             = "RollingBack"
+	Error                   = "Error"
```

Comment:
Why redefine these?  They are in api/v1alpha1/consts.go

Reply:
Will move